### PR TITLE
fix(prepro): Add execption handling to identity and call_function

### PR DESCRIPTION
**Pull request by anna-parker**
resolves https://github.com/loculus-project/loculus/issues/2886

preview URL: https://prepro-patch.loculus.org/

### Summary

Test on preview:
image

Depending on how fast we want this rolled out we can wait on tests or merge this as is. I didn't find any other immediate exception handling issues in the preprocessing_functions.py file, but I added some TODOs for other small issues I found. 

### PR Checklist

- [ ] All necessary documentation has been adapted.
- [ ] The implemented feature is covered by an appropriate test.

### CODE REVIEW

**Review by corneliusroemer — APPROVED** [review]
Skimmed it and looks good

File: preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py [hunk 1]
```diff
@@ -410,9 +441,19 @@ def identity(
...
+                        output_datum = None
+                        warnings.append(invalid_value_annotation(input_datum, output_field, "int"))
                 case "float":
-                    output_datum = float(input_datum)
+                    try:
+                        output_datum = float(input_datum)
+                    except ValueError:
+                        output_datum = None
+                        warnings.append(
+                            invalid_value_annotation(input_datum, output_field, "float")
```
Comment (theosanderson) [inline preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py:455]: Do we definitely want this sort of thing to be just a warning not an error?

Comment (anna-parker) [reply preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py:455]: I think the standard is just an error if the field is required (which will be the case), here we return a warning so the user will see this and then if they ignore it should be submitted without this field (e.g. output_datum=null)

Comment (anna-parker) [reply preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py:455]: Tested and this works

File: preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py [hunk 2]
```diff
@@ -86,6 +111,7 @@ def call_function(
             ],
         )
 
+    # TODO: function returns error even if field is not required, should be warning
```
Comment (theosanderson) [inline preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py:114]:
```suggestion

```

I think with the changes you've now made this is consistent?

File: preprocessing/nextclade/src/loculus_preprocessing/prepro.py [hunk 3]
```diff
@@ -654,18 +655,51 @@ def process_single(
...
+            alignedNucleotideSequences=defaultdict(dict[str, Any]),
+            nucleotideInsertions=defaultdict(dict[str, Any]),
+            alignedAminoAcidSequences=defaultdict(dict[str, Any]),
+            aminoAcidInsertions=defaultdict(dict[str, Any]),
+        ),
+        errors=[
+            ProcessingAnnotation(
+                source=[AnnotationSource(name="unknown", type=AnnotationSourceType.METADATA)],
+                message=(
+                    f"Failed to process submission with id: {id} - please review your submission"
```
Comment (theosanderson) [inline preprocessing/nextclade/src/loculus_preprocessing/prepro.py:674]:
```suggestion
                    f"Failed to process submission with id: {id} - please review your submission "
```

File: preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py [hunk 4]
```diff
@@ -57,6 +64,24 @@ def call_function(
...
+                    errors=[
+                        ProcessingAnnotation(
+                            source=[
+                                AnnotationSource(
+                                    name=output_field, type=AnnotationSourceType.METADATA
+                                )
+                            ],
+                            message=(
+                                f"Internal Error: Function {function_name} did not return "
+                                f"ProcessingResult with input {input_data} and args {args},"
```
Comment (theosanderson) [inline preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py:79]:
```suggestion
                                f"ProcessingResult with input {input_data} and args {args}, "
```

**Review by theosanderson — APPROVED** [review]

**Pull request merged**